src/Symfony/Component/Routing/Loader/ServiceRouterLoader.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Loader;
+
+use Symfony\Component\Config\Loader\Loader;
+use Symfony\Component\Config\Resource\FileResource;
+use Symfony\Component\DependencyInjection\ContainerInterface;
```

Comment:
Another possibility would be to have an abstract class here that would support any container by adding an abstract protected method like `abstract protected function getService($id)` that then could be specialized for the Symfony SC or Pimple for Silex.